pkg/node/routing.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"fmt"
+)
+
+// EncapsulationType is a networking encapsulation type
+//
+// WARNING - STABLE API: This structured is exported to JSON and stored in the
+// kvstore. All changes must be done while guaranteeing backwards
+// compatibility.
+type EncapsulationType string
+
+const (
+	// EncapsulationDisabled indicates to disable encapsulation
+	EncapsulationDisabled EncapsulationType = "disabled"
+
+	// EncapsulationVXLAN indicates to use VXLAN encapsulation mode
+	EncapsulationVXLAN = "vxlan"
```

Comment:
Are the second two consts here taking the correct type? (I think it actually doesn't matter, since consts are placed in as literals, I think, and this type is a string anyway).